### ISSUE TITLE
Crash when following example for arch testing

### ISSUE THREAD

**Issue by marcLeeroy** [pestphp/pest#705]
Hi,

in pest 2.0, Im trying this:

```php
expect('ProjectNamespace\ClassGroup')->not->toUse('DependencyNamespace\Debug');
```

would expect pass or fail but get this

PHPUnit\Architecture\Asserts\Dependencies\ObjectDependenciesDescription::getDocBlockTypeWithNamespace(): Argument #1 ($type) must be of type phpDocumentor\Reflection\Type, null given, called in /vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Methods/Elements/MethodDescription.php on line 76

if i pass the full class name to expect, then it works, but that is not useful.

The class being implemented for all the classes in the namespace has a docblock that says what it uses (@uses)

Am i missing something?

Thanks

**Comment by valkoff** [comment]
I've had the same problem with models in App directory and namespace. I moved them to the appropriate Models directory and then it worked. Now I'm having the same problem and I think it's because of some wrong definition of namespaces across the legacy project I'm working on. 
I think it would be useful to have a check for proper namespacing definitions.
Edit: with more inspection I found the problem is related to wrong definitions of phpdocs, for example I found no typization on an attribute but a textual description. The first word of the description is taken as the type. You can check that too

**Comment by alexanderkroneis** [comment]
I am facing the same issue:

```
  FAILED  Tests\Architecture\DebuggingTest > globals                                                                                                                              TypeError   
  PHPUnit\Architecture\Asserts\Dependencies\ObjectDependenciesDescription::getDocBlockTypeWithNamespace(): Argument #1 ($type) must be of type phpDocumentor\Reflection\Type, null given, called in /var/www/vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Methods/Elements/MethodDescription.php on line 76
```

when following the [documentation](https://pestphp.com/docs/arch-testing#content-tobeused):
```
test('globals')
    ->group('architecture')
    ->expect(['dd', 'dump'])
    ->not->toBeUsed();
```

**Comment by MortenDHansen** [comment]
I have just today upgraded and decided to check if upgrade was successful by copying the example from the documentation like @alexgaal mentions. I get the same error. 

Following the upgrade guide, the only omission on my part was to remove phpunit dependency.

> PHPUnit\Architecture\Asserts\Dependencies\ObjectDependenciesDescription::getDocBlockTypeWithNamespace(): Argument #1 ($type) must be of type phpDocumentor\Reflection\Type, null given, called in /var/www/html/vendor/ta-tikoma/phpunit-architecture-test/src/Asserts/Methods/Elements/MethodDescription.php on line 76

**Comment by eyedroot** [comment]
> with more inspection I found the problem is related to wrong definitions of phpdocs, for example I found no typization on an attribute but a textual description. The first word of the description is taken as the type. You can check that too

@valkoff This was the solution thanks!

**Comment by nunomaduro** [comment]
We need to prepare the plugin for those kind of situations. @fabio-ivona you think you can take a look into this?

**Comment by fabio-ivona** [comment]
@nunomaduro can take a look :+1:

**Comment by nunomaduro** [comment]
@fabio-ivona I believe we will have to do something like this somewhere: https://github.com/pestphp/pest-plugin-arch/blob/8bb489c635b386188bf725b0fd0415262d633903/src/Objects/VendorObjectDescription.php#L25. However, we need a couple of new tests to see exactly what's happening first.

**Comment by fabio-ivona** [comment]
@nunomaduro I think there is an issue here: [MethodDescription.php:75](https://github.com/ta-tikoma/phpunit-architecture-test/blob/1f4466e8faf8d7edd1ae26656f39bf9447c5fa38/src/Asserts/Methods/Elements/MethodDescription.php#L75)

this would fix, I'm opening a PR in `PHPUnit\Architecture` package: https://github.com/ta-tikoma/phpunit-architecture-test/pull/8

```diff
+if ($tag->getVariableName() === $name && $tag->getType() !== null) {
-if ($tag->getVariableName() === $name) {
      $type = $objectMethodsDescription->getDocBlockTypeWithNamespace($tag->getType());
      break;
}
```

**Comment by MortenDHansen** [comment]
I can confirm that fixes it 👍

**Comment by nunomaduro** [comment]
Fixed on v2.2.3.
